NEW Add tip to Alt Text field in Insert Media dialog

Cheddam opened the pull request:
Applies tips to the Alt Text and Title Tooltip fields in the Insert Media dialog, or falls back to descriptions if tips aren't available.

Resolves #983.

Cheddam commented:
@clarkepaul This update matches the wording in the designs you provided - can you confirm you're happy with it?

clarkepaul commented:
Yep I think so unless anything stands out as not coming across well. Feel free to provide a peer review of the text @Cheddam

maxime-rainville commented on code/Forms/ImageFormFactory.php:
```diff
@@ -77,20 +79,42 @@ protected function getFormFieldAttributesTab($record, $context = [])
+
+        $altTextDescription = _t(
+            'SilverStripe\\AssetAdmin\\Controller\\AssetAdmin.AltTextDescription',
+            implode(
```
Isn't it a tiny bit convoluted to use implode to add a space between two strings?

maxime-rainville replied:
The content of the `AltTextDescription` message has changed substantially. We probably need to use a different translation key as any existing translation won't match the new content.

brynwhyman commented:
>I assumed that when you leave the alttext textfield blank, your image should have an empty alt on the front end. The Image Shorcode parser seem to be too clever for its own good and filling that in from the image metadata.

Yeah, @maxime-rainville is correct here. That's what I intended to convey in the issue with: "**1. If alt text is empty the alt attribute should be empty**"

Cheddam commented:
@maxime-rainville @brynwhyman That behaviour will be corrected when [this PR](https://github.com/silverstripe/silverstripe-assets/pull/338) is merged (I thought we'd gotten this across the line a while back, but the tests failed. Working on getting them green now.)

I've addressed the feedback related to this PR, so if you're happy with the state of it please give it a green tick and merge ✅

dnsl48 commented:
Doesn't seem to be covering the AC 1

![peek-asset-admin-1027 webm](https://user-images.githubusercontent.com/9313746/68627695-4cf08400-0543-11ea-8330-dac673c231ce.gif)

Cheddam commented:
@dnsl48 As mentioned in a comment above, AC 1 is covered by [a separate PR](https://github.com/silverstripe/silverstripe-assets/pull/338) which also needs merging.
